src/WebJobs.Script/Binding/ExtensionLoader.cs

```diff
@@ -0,0 +1,224 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Azure.WebJobs.Host;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Binding
+{
+    internal class ExtensionLoader
+    {
+        private readonly TraceWriter _traceWriter;
+        private readonly ILogger _startupLogger;
+        private readonly ScriptHostConfiguration _config;
+
+        // The set of "built in" binding types. These are types that are directly accesible without
+        // needing an explicit load gesture.
+        private static IReadOnlyDictionary<string, string> _builtinBindingTypes = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+        {
+            { "bot", "Microsoft.Azure.WebJobs.Extensions.BotFramework.Config.BotFrameworkConfiguration, Microsoft.Azure.WebJobs.Extensions.BotFramework" },
+            { "sendgrid", "Microsoft.Azure.WebJobs.Extensions.SendGrid.SendGridConfiguration, Microsoft.Azure.WebJobs.Extensions.SendGrid" },
+            { "eventGridTrigger", "Microsoft.Azure.WebJobs.Extensions.EventGrid.EventGridExtensionConfig, Microsoft.Azure.WebJobs.Extensions.EventGrid" }
+        };
+
+        private static IReadOnlyDictionary<string, string> _builtinScriptBindingTypes = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
```

Comment:
can we get a comment on the difference between _builtinBindingTypes and _builtinScriptBindingTypes @MikeStall

Reply:
Yeah, wasn't clear to me either

Reply:
_builtinBindingTypes are built on BYOB. This is the correct long-term approach. 
_builtinScriptBindingTypes are not built on BYOB - they have ScriptBindingProviders (which is glue code we aim to get rid of.)

---
In reply to: [138433824](https://github.com/Azure/azure-webjobs-sdk-script/pull/1888#discussion_r138433824) [](ancestors = 138433824)

Reply:
I'll add a code comment